packages/prompts/src/index.ts

```diff
@@ -929,6 +929,85 @@ export const spinner = ({
 	};
 };
 
+export type ProgressOptions = SpinnerOptions & {
+	style?: 'light' | 'heavy' | 'block';
+	max?: number;
+	size?: number;
+};
+export interface ProgressResult {
+	start(msg?: string): void;
+	stop(msg?: string, code?: number): void;
+	message(msg?: string): void;
+	advance(step?: number, msg?: string): void;
+	readonly isCancelled: boolean;
+}
+
+export function progress({
+	style = 'heavy',
+	max = 100,
+	size = 40,
+	indicator = 'dots',
+	onCancel,
+	output = process.stdout,
+	cancelMessage,
+	errorMessage,
+	input,
+}: ProgressOptions = {}): ProgressResult {
+	const spin = spinner({
+		onCancel,
+		cancelMessage,
+		errorMessage,
+		indicator,
+		output,
+		input,
+	});
+	const progressChar: Record<NonNullable<ProgressOptions['style']>, string> = {
```

Comment:
can we move this outside the function since it is static?

we should probably do the same for spinner frames in a separate PR one day

Reply:
I also renamed it to match the rest of the static values in the file.